### ISSUE TITLE
Namespaced url pattern is repeated without namespace, collides with other patterns

### ISSUE THREAD

**Issue by Bouke** [noirbizarre/django.js#10]
I've got the following url patterns defined:

```
api:account:account-detail # api url
account:account-detail # web url
```

However, in `init.js` the url patterns are repeated with its namespaces stripped off. Now when I want to use the `account:account-detail` url pattern, it wrongfully uses the `api:account:account-detail` url pattern, as it comes first.

```
api:account:account-detail # api url
account:account-detail # api url
account-detail # api url
account:account-detail # web url
account-detail # web url
```

In `init.js` I only need the exact url pattern names, not with their namespaces stripped off. How can I do this?

**Comment by noirbizarre** [comment]
I will need some more details.
This case should be tested and shouldn't happen (see [djangojs/tests/test_urls.py](https://github.com/noirbizarre/django.js/blob/master/djangojs/tests/test_urls.py#L71,L103)).

How do you declare your url patterns ? Maybe this case is not covered.

**Comment by noirbizarre** [comment]
So, have you checked your way of declaring namespaces in url patterns.
Is it the same as in the test suit ?

Tell me if you want me to fix this issue.

**Comment by Bouke** [comment]
So I've traced back the problem and it seems to be related to the ordering of including the urls. So these are my urls:

`api/urls.py`:

```
urlpatterns = patterns('', url(r'', include('api.accounts.urls', namespace='accounts')))
```

`api/accounts/urls.py`:

```
urlpatterns = patterns('', url(r'...', view, name='account-detail'))
```

`accounts/urls.py`:

```
urlpatterns = patterns('', url('...', view, name="account-detail"))
```

With `ROOT_URLCONF` like this, the `accounts:account-detail` is matched to the one in `accounts/urls.py` (expected):

```
urlpatterns = patterns('',
    url(r'^api/', include('api.urls', namespace='api')),
    url(r'^', include('accounts.urls', namespace='accounts')),
)
```

With `ROOT_URLCONF` like this, the `accounts:account-detail` is matched to the one in `api/accounts/urls.py` (unexpected):

```
urlpatterns = patterns('',
    url(r'^', include('accounts.urls', namespace='accounts')),
    url(r'^api/', include('api.urls', namespace='api')),
)
```

So regarding the testcase, this situation is not covered yet. You would need a `A:B:C` and a `B:C` route, where `B:C` should always match to `B:C` regardless of ordering in `ROOT_URLCONF`. Thanks!

**Comment by noirbizarre** [comment]
Should be fixed in release 0.6.4 (just published)
